nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/P2BrowseNodeGenerator.java

```diff
@@ -42,49 +38,51 @@
 public class P2BrowseNodeGenerator
     implements BrowseNodeGenerator
 {
-  private static final String REMOTE_URL_PREFIX_REGEX = "(" + HTTP_NXRM_PREFIX + "|" + HTTPS_NXRM_PREFIX + ")";
+  private final Set<String> knownSubDirectories;
+
+  @Inject
+  public P2BrowseNodeGenerator() {
+    Set<String> knownFirstSegments = new HashSet<>();
```

Comment:
As far as I remember, it was code which aggregate feature, plugin and binary in one component if it's same version, here was the ticket where it was implemented https://issues.sonatype.org/browse/NEXUS-22157 , you can take a look on https://issues.sonatype.org/secure/attachment/97502/97502_Screen+Shot+2019-12-20+at+11.05.33.png. Is it possible to keep this feature?

Reply:
> As far as I remember, it was code which aggregate feature, plugin and binary in one component if it's same version, here was the ticket where it was implemented https://issues.sonatype.org/browse/NEXUS-22157 , you can take a look on https://issues.sonatype.org/secure/attachment/97502/97502_Screen+Shot+2019-12-20+at+11.05.33.png. Is it possible to keep this feature?

Its possible but I removed it because I don't believe that behaviour to be correct. IMO the identifiers should be treated as opaque and we shouldn't infer connections between two bundles based on arbitrary rules.